### PR TITLE
Add support for support arbitrary key:value pairs in ListGenerator

### PR DESCRIPTION
**Pull request by maruina**
Fix https://github.com/argoproj-labs/applicationset/issues/108

### CODE REVIEW

File: examples/list.yaml [hunk 1]
```diff
@@ -9,15 +9,21 @@ spec:
       elements:
       - cluster: engineering-dev
         url: https://1.2.3.4
+        values:
+          project: dev
       - cluster: engineering-prod
         url: https://2.4.6.8
+          values:
+            project: prod
       - cluster: finance-preprod
```
Comment (mgoodness) [inline examples/list.yaml:18]: L16-17: fix indentation

**Review by mgoodness — CHANGES_REQUESTED** [review]

File: examples/list.yaml (hunk 1 above)
Comment (maruina) [reply examples/list.yaml:18]: Good catch, fixed. I also changed to use single quote instead of double quotes.

**Review by mgoodness — APPROVED** [review]
LGTM

**Comment by OmerKahani** [comment]
LGTM. Thanks @maruina

**Review by OmerKahani — APPROVED** [review]

**Comment by maruina** [comment]
I don't know if this PR covers everything in https://github.com/argoproj-labs/applicationset/issues/98 but I added a couple of tests. It might be a good start.

**Review by jgwest — APPROVED** [review]
LGTM too, thanks @maruina!

**Comment by yogeek** [comment]
Hello @maruina @jgwest 
Nice to see this new feature.
I would like to clarify 2 points please :

- the documentation does not seem to reflect this new possibility in the List generator... do you confirm ?
- is that a particular reason to keep `cluster` and `url` as mandatory fields in the list generator ?

**Comment by jgwest** [comment]
Hi @yogeek, 

> the documentation does not seem to reflect this new possibility in the List generator... do you confirm ?

Confirmed, I have opened https://github.com/argoproj-labs/applicationset/issues/273 for this.

> is that a particular reason to keep cluster and url as mandatory fields in the list generator ?

Nope, see the discussion on https://github.com/argoproj-labs/applicationset/issues/162 re: getting rid of these as mandatory fields.

**Comment by yogeek** [comment]
Hi @jgwest :-)

Great, thank you !

**Pull request merged**